src/Middleware/Rewrite/test/IISUrlRewrite/ServerVariableTests.cs

```diff
@@ -53,7 +102,24 @@ private RewriteContext CreateTestHttpContext()
             context.Request.Headers[HeaderNames.Referer] = "referer";
             context.Request.Headers[HeaderNames.UserAgent] = "useragent";
             context.Request.Headers[HeaderNames.Connection] = "connection";
-            return new RewriteContext { HttpContext = context };
+
+            return context;
+        }
+
+        private RewriteContext CreateTestRewriteContext(HttpContext context = null)
+        {
+            return new RewriteContext { HttpContext = context ?? CreateTestHttpContext() };
+        }
+
+        private IServerVariablesFeature CreateTestServerVariablesFeature(Dictionary<string, string> variables)
+        {
+            var mockServerVariablesFeature = new Mock<IServerVariablesFeature>();
```

Comment:
I'd prefer just creating a concrete `TestServerVariablesFeature` class that takes a dictionary and exposes it via the accessor. It's trivial and means we don't need more `Mock`s. In general, I lean towards using actual types rather than more dynamic code gen :)

```csharp
public class TestServerVariablesFeature: IServerVariablesFeature
{
    private readonly Dictionary<string, string> _variables;

    public TestServerVariablesFeature(IDictionary<string, string> variables)
    {
        _variables = variables;
    }

    public string this[string variableName]
    {
        get => _variables[variableName];
        set => _variables[variableName] = value;
    }
}
```